### PR TITLE
DM-42631: Add CC-IN2P3 (Slurm) as a target site

### PR DESCRIPTION
**Pull request by airnandez**
## Checklist

- [x] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

**Review by PaulPrice — APPROVED** [review]
Third commit should be squashed into the second.
Need to add docs for new feature.
Please consider changing `get_parsl_config` to supplement/override rather than constructing from scratch.

File: python/lsst/ctrl/bps/parsl/workflow.py [hunk 1]
```diff
@@ -65,11 +69,14 @@ def get_parsl_config(config: BpsConfig) -> parsl.config.Config:
         Parsl configuration.
     """
     site = SiteConfig.from_config(config)
-    executors = site.get_executors()
-    retries = get_bps_config_value(site.site, "retries", int, 1)
-    monitor = site.get_monitor()
+    if parsl_config := site.get_parsl_config():
+        return parsl_config
```
[the remark this answers is not in the document]
Comment (airnandez) [reply python/lsst/ctrl/bps/parsl/workflow.py]: Added a method `get_parsl_config()` to base class `SiteConfig`, overwritable by subclasses. This seems to me cleaner now.

File: python/lsst/ctrl/bps/parsl/workflow.py [hunk 2]
```diff
@@ -65,11 +69,14 @@ def get_parsl_config(config: BpsConfig) -> parsl.config.Config:
         Parsl configuration.
     """
     site = SiteConfig.from_config(config)
-    executors = site.get_executors()
-    retries = get_bps_config_value(site.site, "retries", int, 1)
-    monitor = site.get_monitor()
```
[the remark this answers is not in the document]
Comment (airnandez) [reply python/lsst/ctrl/bps/parsl/workflow.py:70]: Reverted. I kept the previous style.

File: python/lsst/ctrl/bps/parsl/site.py [hunk 3]
```diff
@@ -197,3 +198,17 @@ def get_monitor(self) -> MonitoringHub | None:
             logging_endpoint="sqlite:///"
             + get_bps_config_value(self.site, "monitorFilename", str, "monitor.sqlite"),
         )
+
+    def get_parsl_config(self) -> parsl.config.Config | None:
+        """Get Parsl configuration for this site.
+
+        This method allows concrete subclasses to override this method to
+        provide a a Parsl configuration specific for the site. If not
```
[the remark this answers is not in the document]
Comment (airnandez) [reply python/lsst/ctrl/bps/parsl/site.py]: Fixed.

**Comment by airnandez** [comment]
@PaulPrice Many thanks for your feedback. I think I implemented your recommendations. You may want to check the final result before I merge.

File: python/lsst/ctrl/bps/parsl/sites/ccin2p3.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        config : `parsl.config.Config`
+            The configuration to be used to initialize Parsl for this site.
+        """
+        executors = self.get_executors()
+        monitor = self.get_monitor()
+        retries = get_bps_config_value(self.site, "retries", int, 1)
+        # Path to Parsl run directory. The default set by Parsl is
+        # 'runinfo' which is not explicit enough for end users given that
+        # we are using BPS + Parsl + Slurm to execute a workflow.
+        run_dir = get_bps_config_value(self.site, "run_dir", str, "parsl_runinfo")
```
Comment (PaulPrice) [inline python/lsst/ctrl/bps/parsl/sites/ccin2p3.py:236]: Oh, this is a nice feature! It might be good to hoist this into the parent class.

**Comment by PaulPrice** [comment]
Thanks, it looks good. It would be helpful if you could clean up the commit history some, as it looks like there are more commits than changes in the end.

**Comment by airnandez** [comment]
I tried to clean the history but failed. Apologies for that. It became a mess: my knowledge of git was not enough to sort things out in a reasonable amount of time, so I preferred to get the code merged as is. Lesson learned for future contributions, hopefully.

**Pull request merged**